src/vsphere_cpi/lib/cloud/vsphere/drs_rules/drs_rule.rb

```diff
@@ -71,11 +71,11 @@ def update_rule(rule_key)
       add_anti_affinity_rule(VimSdk::Vim::Option::ArrayUpdateSpec::Operation::EDIT, rule_key)
     end
 
-    def create_vm_host_affinity_rule(vm_group_name, host_group_name)
+    def create_vm_host_affinity_rule(vm_group_name, host_group_name, rule_type)
       vm_host_rule_info = VimSdk::Vim::Cluster::VmHostRuleInfo.new
       vm_host_rule_info.enabled = true
-      # This is a HARD AFFINITY rule.
-      vm_host_rule_info.mandatory = true
+      # Check Hard or Soft Affinity
+      vm_host_rule_info.mandatory = (rule_type == VSphereCloud::Resources::Cluster::CLUSTER_VM_HOST_RULE_MUST) ? true : false
```

Comment:
Unless `rule_type` is a very magic type then the ternary operator here is unnecessary.